src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -816,8 +816,35 @@ void AbstractWriteLog<I>::write(Extents &&image_extents,
 
   ceph_assert(m_initialized);
 
+  Extents splitted_image_extents;
+  uint64_t max_extent_size = get_max_extent();
+  if (max_extent_size != 0) {
+    for (auto extent : image_extents) {
+      if (extent.second > max_extent_size) {
+	uint64_t off = extent.first;
+	uint64_t extent_bytes = extent.second;
+	int len = ceil(extent_bytes * 1.0 / max_extent_size);
```

Comment:
Nit: why use floating point math here? Why not just loop and append `std::min(max_extent_size, extent_bytes)` bits until `extent_bytes == 0`?